pkg/controllers/raycluster_webhook.go

```diff
@@ -123,6 +127,32 @@ func (w *rayClusterWebhook) Default(ctx context.Context, obj runtime.Object) err
 		}
 	}
 
+	hasSecurityLabels, err := namespaceHasSecurityLabels(ctx, w.Client, rayCluster.Namespace)
+	if err != nil {
+		rayclusterlog.Error(err, "Failed to check namespace resource labels")
+		return err
+	}
+	if hasSecurityLabels {
```

Comment:
Why not adding the security context unconditionally?

Reply:
I actually wasn't sure if that's what we wanted, hence made it conditional. On a security point a view, you're right, I think we should always have this securityContext enabled. Should I make it unconditional, and also add to the `ValidatingWebhook` to ensure these are set?

Reply:
Yes, I think that should be added unconditionally. I'm not sure how it'll behave on non OpenShift clusters. We may make it only for OpenShift if it has to.

Reply:
Hmm interesting point. I'll make the changes and check the e2e tests. This should be sufficient to confirm its behaviour. Thanks Antonin!